src/ripple/protocol/Rate.h

```diff
@@ -43,7 +44,13 @@ struct Rate
     Rate (std::uint32_t rate)
         : value (rate)
     {
-        assert (rate != 0);
+    }
+
+    Rate&
+    operator=(std::uint32_t rate)
```

Comment:
Also, for the long term, I feel like this assignment operator allows folks to be a bit too cavalier mixing `Rate` and `uint32`.  Perhaps renaming this method `set()` in the future (not this pull request) would be a good idea.